src/gululog_idx.erl

```diff
@@ -359,6 +379,43 @@ file_sync_close(Fd) ->
   ok = file:sync(Fd),
   ok = file:close(Fd).
 
+%% @private Delete index and segment files for truncate
+-spec truncate_after_delete_do([{logid(), {segid(), position()}}], cache(), dirname()) ->
+  [{filename(), filename()}].
+truncate_after_delete_do([], _Tid, _Dir) ->
+  [];
+truncate_after_delete_do(DeleteList, Tid, Dir) ->
+  [begin
+     ets:delete(Tid, LogIdX),
+     ok = file:delete(SegFileD = gululog_name:mk_seg_name(Dir, SegIdX)),
```

Comment:
not a good idea to delete/truncate seg files in this module.
it's probably better to be done in gululog_topic like this:

-spec truncate_inclusive(topic(), logid()) -> {topic(), [filename()]}.
truncate_inclusive(#topic{dir = Dir, idx = Idx, cur = Cur} = Topic, LogId) ->
  {SegId, SegPosition} = gululog_idx:locate(Dir, Idx, LogId),
  %% delete the truncated cache entries
  %% close writer fd
  %% delete the files > SegId
  %% truncate the file = SegId
  %% re-open the writer fd
  {NewIdx, DeletedIdxFiles} = gululog_idx:truncate(Idx, SegId, LogId),
  %% close writer fd
  %% delete the files > SegId
  %% truncate the file = SegId
  %% re-open the writer fd
  {NewCur, DeletedSegFiles} = gululog_w_cur:truncate(Cur, SegId, SegPosition),
  {Topic#{ idx = NewIdx
         , cur = NewCur
         }, DeletedIdxFiles ++ DeletedSegFiles}.